File: Data Structures & Algorithms/binary-tree-diameter/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root) {

        let max = 0;

        function find(root){
            if (root == null) return 0;

            let left = find(root.left);
            let right = find(root.right);

            max = Math.max(max, left+right);
            return 1 + Math.max(left,right);
        };

        find(root);
        return max;

    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        left = 0
        right = 1
        while right < len(nums):
            if nums[left] == nums[right] and right - left <= k:
                return True
            if right - left <= k:
                right += 1
            else:
                left += 1


        return False